### PR TITLE
test : old version(< 0.7.1) client test case for multi-version protocol 

### LINKED ISSUE #6694: Multi-version protocol compatibility feature requires a test collection for < 0.7.1
## Why you need it?
 Is your feature request related to a problem? Please describe in details
https://github.com/apache/incubator-seata/pull/6226
Multi-version protocol compatibility feature requires a test collection for < 0.7.1

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

### PR DESCRIPTION
**Pull request by Bughue**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
 old version(< 0.7.1) client test case for Multi-version protocol 

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/apache/incubator-seata/issues/6694

### CODE REVIEW

File: test-old-version/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.seata</groupId>
+            <artifactId>seata-rm</artifactId>
+            <version>0.6.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.seata</groupId>
+            <artifactId>seata-test</artifactId>
+            <version>2.1.0-SNAPSHOT</version>
```
Comment (funky-eyes) [inline test-old-version/pom.xml]:
```suggestion
            <version>${revision}</version>
```

File: test-old-version/src/test/resources/basic-test-context.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		<property name="testWhileIdle" value="true" />
+		<property name="timeBetweenEvictionRunsMillis" value="60000" />
+		<property name="minEvictableIdleTimeMillis" value="25200000" />
+		<property name="removeAbandoned" value="true" />
+		<property name="removeAbandonedTimeout" value="1800" />
+		<property name="logAbandoned" value="true" />
+		<property name="filters" value="mergeStat" />
+	</bean>
+
+    <bean id="dataSourceProxy" class="org.apache.seata.rm.datasource.DataSourceProxy">
```
Comment (funky-eyes) [inline test-old-version/src/test/resources/basic-test-context.xml]: 为什么要加这些测试？理论上只要测试request是否能得到预期内的response不是吗？
Why add these tests? In theory, as long as the test request can get the expected response, right?

File: test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * the type ConfigurationTestHelper
+ **/
+public class ConfigurationTestHelper {
```
Comment (funky-eyes) [inline test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java]: 为什么不把这个类放到seata-test中复用
Why not reuse this class in seata-test?

Comment (Bughue) [reply test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java]: 这个类在seata-test了，但是他是在test包里的，这样共用不了吧，还要另外搞plugin什么的

Comment (funky-eyes) [reply test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java]: > 这个类在seata-test了，但是他是在test包里的，这样共用不了吧，还要另外搞plugin什么的

类似弄一个seata-test-common？
Like getting a seata-test-common?

**Review by funky-eyes — COMMENTED** [review]
seata-parent 中没有添加该模块，github上ci压根没有跑起来
The module was not added to Ceata Parente, and Gitub didn't run at all last time

File: test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java (hunk 3 above)
Comment (Bughue) [reply test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java]: 我也考虑过，但这个common jar里就只有一个类了吗？
I've considered it, but is there just one class in this common jar?

Comment (funky-eyes) [reply test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java]: > 我也考虑过，但这个common jar里就只有一个类了吗？ I've considered it, but is there just one class in this common jar?

你可以梳理下，看看有没有什么其他可以复用的类和方法
You can sort it out to see if there are any other classes and methods that can be reused.

Comment (Bughue) [reply test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java]: 几乎没有，原因是他们要测的包不一样，一个是测旧版本的io.seata，新的是apache的，所以他们真正共用的就这个类，其他的都是代码很相似但是引用不一样。
Almost nothing else.the reason is that they want to test different packages, one is to test the old version of io.seata, the new one is apache, so they really share this class, the other are very similar code but references are not the same

Comment (Bughue) [reply test-old-version/src/test/java/io/seata/core/rpc/netty/ConfigurationTestHelper.java:31]: 又想了一下，其实这里是为了改mock的端口，要不把他移动到Mock那个包？

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**